<launch>

    <!-- <set_env name="RCUTILS_CONSOLE_OUTPUT_FORMAT" value="{function_name}(): {message} "/> -->
    <set_env name="RCUTILS_CONSOLE_OUTPUT_FORMAT" value="{time}: {message} "/>
    <set_env name="RCUTILS_COLORIZED_OUTPUT" value="1"/>
    
    <!-- 変数 -->
    <arg name="name_space" default=""/> <!-- config/{}.yaml内のnamespaceとそろえる必要あり -->
    <arg name="name_pkg" default="dynamixel_handler"/>

    <node pkg="$(var name_pkg)" exec="dynamixel_handler" name="dynamixel_handler" namespace="$(var name_space)" output="screen">
        <!-- 細かいパラメータの設定 -->
        <param from="$(find-pkg-share $(var name_pkg))/config/config_dynamixel_handler.yaml"/>
        <!-- <param from="$(find-pkg-share $(var name_pkg))/config/config_dynamixel_handler-fast.yaml"/> -->

        <!-- トピックの設定 -->
        <!-- 書き込み -->
        <remap from="dynamixel/commands/x"   to="dynamixel/commands/x"/>
        <remap from="dynamixel/command/x/pwm_control"                   to="dynamixel/command/x/pwm_control"/>
        <remap from="dynamixel/command/x/current_control"               to="dynamixel/command/x/current_control"/>
        <remap from="dynamixel/command/x/velocity_control"              to="dynamixel/command/x/velocity_control"/>
        <remap from="dynamixel/command/x/position_control"              to="dynamixel/command/x/position_control"/>
        <remap from="dynamixel/command/x/extended_position_control"     to="dynamixel/command/x/extended_position_control"/>
        <remap from="dynamixel/command/x/current_base_position_control" to="dynamixel/command/x/current_base_position_control"/>
        <remap from="dynamixel/command/status" to="dynamixel/command/status"/>
        <remap from="dynamixel/command/goal"   to="dynamixel/command/goal"/>
        <remap from="dynamixel/command/gain"   to="dynamixel/command/gain"/>
        <remap from="dynamixel/command/limit"  to="dynamixel/command/limit"/>
        <remap from="dynamixel/command/extra"  to="dynamixel/command/extra"/>
        <remap from="dynamixel/shortcut" to="dynamixel/shortcut"/>

        <!-- 読み込み -->
        <remap from="dynamixel/states" to="dynamixel/states"/>
        <remap from="dynamixel/state/status"  to="dynamixel/state/status"/>
        <remap from="dynamixel/state/present" to="dynamixel/state/present"/>
        <remap from="dynamixel/state/goal"    to="dynamixel/state/goal" />
        <remap from="dynamixel/state/gain"    to="dynamixel/state/gain" />
        <remap from="dynamixel/state/limit"   to="dynamixel/state/limit"/>
        <remap from="dynamixel/state/error"   to="dynamixel/state/error"/>
        <remap from="dynamixel/state/extra"   to="dynamixel/state/extra"/>
        <remap from="dynamixel/debug" to="dynamixel/debug"/>
    </node>

</launch>